datetime/src/test/java/com/fasterxml/jackson/datatype/jsr310/deser/LocalDateDeserTest.java

```diff
@@ -57,20 +57,29 @@ public ShapeWrapper() { }
     }
 
     final static class StrictWrapper {
-        @JsonFormat(pattern="yyyy-MM-dd",
+        @JsonFormat(pattern="yyyy-MM-dd G",
```

Comment:
Instead of changing this test format, we should keep this one 'as-is', and then add entirely new tests (as you have done) for the 'date-with-era' test case.

Based on this SO post, most users will probably default to using 'yyyy', so there should be test coverage for that scenario. Thus, recommend having three Wrapper classes:

 @JsonFormat(pattern="yyyy-MM-dd" // original, leaving as-is (year-of-era, but lacks era)

 @JsonFormat(pattern="yyyy-MM-dd G" // new, improved 'year-of-era', with era

@JsonFormat(pattern="uuuu-MM-dd", // new 'year'

https://stackoverflow.com/questions/41177442/uuuu-versus-yyyy-in-datetimeformatter-formatting-pattern-codes-in-java

Reply:
I am a little unsure about the 'yyyy-MM-dd G' format after reading the StackOverflow post. Maybe we can trim this down to just 'yyyy-MM-dd' (era ambiguous, but typical usage) and 'uuuu-MM-dd' (no ambiguity). 

> We should also see that using eras (symbol "G") in context of ISO is in general dangerous if we consider historic dates. If "G" is used with "u" then both fields are unrelated to each other. And if "G" is used with "y" then the formatter is satisfied but still uses proleptic gregorian calendar when the historic date mandates different calendars and date-handling.

However, in another answer in the same post, they make this point, so probably useful to have both:

> Conclusion: If you use y you should also use G. Since G is rarely used, the correct year symbol is u, not y, otherwise a non-positive year will show incorrectly.

The problems seem to be due to the fact that the designers of [JSR-310](https://jcp.org/en/jsr/detail?id=310) decided to use the [CLDR ](https://en.wikipedia.org/wiki/Common_Locale_Data_Repository)conventions, but:

> .. this decision to follow CLDR was not quite consistent because JSR-310 had also introduced new pattern symbols which didn't and still don't exist in CLDR

Reply:
Yes, thanks @kupci for your reply, I agree with you to have three Wrapper classes you mentioned above and leave the original test case there. 
But for the `yyyy-MM-dd`, I assume the original test case want to test an invalid date("2019-11-31" is invalid) rather an invalid format:
```java
@Test(expected = InvalidFormatException.class)
public void testStrictCustomFormat() throws Exception
{
    /*StrictWrapper w =*/ MAPPER.readValue("{\"value\":\"2019-11-31\"}", StrictWrapper.class);
}
```
For any valid date in this test case is invalid format. So would you mind me to change the `2019-11-31` to `2019-11-30` and cover the `invalid date case` in another test case?

Reply:
Yes, I see your point now, the original test didn't actually test what it was intended to test, or should have tested.